### DISCUSSION TITLE
Enter to open file

### DISCUSSION THREAD

**Discussion by Lance929** [pqrs-org/Karabiner-Elements#4067] [Q&A]
There are many JSON codes to open files or folders using Enter/Return (e.g. Use Return to Open).  However I noticed that when I shift focus to the menu (Ctrl+F2) or dock (Ctrl+F3) and press enter the menu does not open or the dock app does not open, in addition menu items do not activate when I press enter, e.g. when I press enter to shut down my mac, the shut down window does not open, and the highlighted menu item moves away from shut down menu item.  What I would like is to have a JSON code that still opens a file/folder only when it is selected.  However when focus is shifted away from the file/folder I would like the enter/return key to activate those items that are in focus, e.g. dock items, menu items, contextual menu items, window options.  Thanks for any help.  

Lance

**Comment by tekezo** [comment]
This kind of fine control isn't possible, which is a limitation of Karabiner-Elements. For that reason, directly mapping Enter to Command+O is not highly recommended. It would be better to add a little extra logic, such as mapping Enter to Command+O only when Enter is held down.
